bigchaindb/backend/mongodb/connection.py

```diff
@@ -129,6 +206,10 @@ def initialize_replica_set(host, port, connection_timeout, dbname, ssl, login, p
     else:
         _wait_for_replica_set_initialization(conn)
         logger.info('Initialized replica set')
+    finally:
+        if conn is not None:
+            logger.info('Closing initial connection to MongoDB')
+            conn.close()
```

Comment:
We should probably refactor https://github.com/bigchaindb/bigchaindb/pull/1510/files#diff-9e7a185b21f57863613b664d55961cc6R204 or else the `finally` will not be reached if the replicaset is already initialized